index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
@pedromrb That discussion has to do with how to use the decorator, which doesn't apply here.. we are just offering a decorator. People might use the decorator as a normal javascript function or with `@`.. it really is up to them. For instance, in our projects at moxy, we do not use decorators yet because they were not approved by TC39. Instead we use them using "vanilla" javascript

```
export default known.decorator()(myObject);
```

@sindresorhus one thing though, is that #1 is related to classes, not plain objects. This means that `known` must be applied to the instance and not to the constructor/proptotype properties. What do you suggest?